voting_ml/feature_selection.py

```diff
@@ -81,17 +82,16 @@ def split_data(self, X, y, input_test_size=0, input_random_state=None):
 
         return data_dict
             
-    def ftsel_chi2(self, data, KBest = 'all', label=None, input_test_size=0, input_random_state=None):
+    def ftsel_chi2(self, data_dict, KBest = 'all'):
         
-        X, y = self.separate_ft_label(data,label)
-        N = X.shape[0]
-
-        data_dict = self.split_data(X, y, input_test_size, input_random_state)
-
-        X_train = data_dict["X_train"]
-        X_test = data_dict["X_test"]
-        y_train = data_dict["y_train"]
-        y_test = data_dict["y_test"]
+        data = data_dict["data"].astype(str)
+        X_train = data_dict["X_train"].astype(str)
+        X_test = data_dict["X_test"].astype(str)
+        y_train = data_dict["y_train"].astype(str)
+        y_test = data_dict["y_test"].astype(str)
```

Comment:
Why are we casting all of these to strings? The majority of the data is not strings.

Reply:
This is done in many places throughout this file

Reply:
This is because the ordinal encoder apparently needs the data to be of a string format. I am using ordinal encoder because it was easier to use given that some part of our data is strings. We can also use your method and convert everything to integers. At the time it seemed like an easier way and because I was mainly focussed on getting some preliminary results. We can switch to your method of converting everything to integers first and then using the data for feature selection.

Reply:
I was using it also because there are some NA values I wanted to use and not just discard. It was easier to just use the data as string and use ordinal encoder. It still looks like a reasonable option to me as it is easier to handle (because we don't have to assign a number ourselves) and in the end we can get the data in the original format. I see the advantages of converting everything beforehand as we know the transformations however we can also store the transformations from these encoders as well. Let me know what you think.

Reply:
I don't think its a huge deal which one we pick to get the features we want, because ultimately those are questions and they are strings anyway. We will just have to re-query the poll data after we get our chosen features/questions (outside of the `FeatureSelection` class). Since the ordinal encoder needs strings, I would probably just leave those as strings.